### PR TITLE
LRQA-36520 | Master

### PR DESCRIPTION
**Pull request by acbbennett**
https://issues.liferay.com/browse/LRQA-36520

### CODE REVIEW

**Comment by austinchiang** [comment]
@acbbennett Can you also submit on 7.0.x? All new tests should be written on both versions. Please make sure they pass too. Thanks!

File: portal-web/test/functional/com/liferay/portalweb/macros/AdaptiveMedia.macro [hunk 1]
```diff
@@ -212,6 +212,7 @@
 			</else>
 		</if>
 
+		<execute function="AssertClick" locator1="Button#SAVE" value1="Save" />
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/macros/AdaptiveMedia.macro]: Hey @acbbennett Why did you add this line? This is covered in `PortletEntry#save`

**Comment by acbbennett** [comment]
@austinchiang re: "Why did you add this line? This is covered in PortletEntry#save"
PortletEntry#save assumes success; this macro needs a variant for the results of clicking with no dimension values. Leaving the full macro would take the same number of steps.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 2]
```diff
@@ -237,6 +237,55 @@
...
+			<var name="panel" value="Control Panel" />
+			<var name="portlet" value="Adaptive Media" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#addImageResolution">
+			<var name="imageName" value="testResolution" />
+			<var name="maxHeight" value="" />
+			<var name="maxWidth" value="" />
+			<var name="noDimensions" value="true" />
+		</execute>
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: @acbbennett please assert that the image resolution is actually added (think of the MB test you wrote for the training assessment).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 3]
```diff
@@ -452,6 +501,36 @@
...
+
+		<execute macro="AdaptiveMedia#editImageResolution">
+			<var name="imageName" value="testResolution" />
+			<var name="imageNameEdit" value="testResolutionEdit" />
+			<var name="maxHeightEdit" value="" />
+			<var name="maxWidthEdit" value="" />
+			<var name="noDimensions" value="true" />
+		</execute>
+	</command>
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: Same here. Please assert there is no entry.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 4]
```diff
@@ -490,6 +569,50 @@
...
+			<var name="maxWidth" value="1000" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#editImageResolution">
+			<var name="imageName" value="testResolutionHeight" />
+			<var name="imageNameEdit" value="testResolutionHeightEdit" />
+			<var name="maxWidthEdit" value="" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#viewImageResolutionTableEntry" >
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: extra space

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 5]
```diff
@@ -490,6 +569,50 @@
...
+			<var name="maxWidth" value="1000" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#editImageResolution">
+			<var name="imageName" value="testResolutionWidth" />
+			<var name="imageNameEdit" value="testResolutionWidthEdit" />
+			<var name="maxHeightEdit" value="" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#viewImageResolutionTableEntry" >
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: extra space

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 6]
```diff
@@ -452,6 +501,36 @@
...
+			<var name="portlet" value="Adaptive Media" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#addImageResolution">
+			<var name="imageName" value="testResolution" />
+			<var name="maxHeight" value="1000" />
+			<var name="maxWidth" value="1000" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#viewImageResolutionTableEntry" >
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: extra space

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 7]
```diff
@@ -237,6 +237,55 @@
...
+		</execute>
+
+		<execute macro="AdaptiveMedia#addImageResolution">
+			<var name="description" value="Image Resolution Height" />
+			<var name="imageName" value="testResolutionHeight" />
+			<var name="maxHeight" value="1000" />
+			<var name="maxWidth" value="" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#viewImageResolutionTableEntry" >
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: extra space

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase [hunk 8]
```diff
@@ -237,6 +237,55 @@
...
+		</execute>
+
+		<execute macro="AdaptiveMedia#addImageResolution">
+			<var name="description" value="Image Resolution Width" />
+			<var name="imageName" value="testResolutionWidth" />
+			<var name="maxHeight" value="" />
+			<var name="maxWidth" value="1000" />
+		</execute>
+
+		<execute macro="AdaptiveMedia#viewImageResolutionTableEntry" >
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/adaptivemedia/AdaptiveMedia.testcase]: extra space

File: portal-web/test/functional/com/liferay/portalweb/macros/AdaptiveMedia.macro [hunk 9]
```diff
@@ -213,7 +219,16 @@
...
 
 		<execute function="AssertClick" locator1="Button#SAVE" value1="Save" />
-		<execute macro="PortletEntry#save" />
+
+		<if>
+			<isset var="noDimensions" />
+			<then>
+				<execute function="AssertTextEquals" locator1="Message#REQUIRED_INFO" value1="At least one value is required." />
+			</then>
+		<else>
```
Comment (austinchiang) [inline portal-web/test/functional/com/liferay/portalweb/macros/AdaptiveMedia.macro]: indent

**Comment by austinchiang** [comment]
SF for @acbbennett: Please see my comments and resend. Thanks!
